JS/Lesson16/tugba/script.js

```diff
@@ -60,83 +60,137 @@ console.log(newArray);
-function getRandomNumber () {
-return Math.floor(Math.random()*10)+1;
-} 
-
-let iteration = 1;
-let randomNumber = getRandomNumber();
-let randomNumbers = [randomNumber];
-
-while(randomNumber <= 8){
-randomNumber = getRandomNumber();
-randomNumbers.push(randomNumber);
-iteration++;
+
+let totalIterationsNumber = 0;
+let randomNumbers = 0 [randomNumber];
+
+while (randomNumber <= 8) {
+	randomNumber = Math.floor(Math.random() * 10) + 1;
+	totalIterationsNumber++;
 }
+console.log("Exercise 8: ", 'Total number of iterations: ', totalIterationsNumber);
 
-console.log('Iterations:', iteration, 'random number: ', randomNumber, 'Numbers: ', randomNumbers);
 
-console.log('- - - - - - - - - - - - - - - - - ');
+console.log("- - - - - - - - - - - - - - - - - ");
 // 9. Write a do-while loop that counts from 1 to 10 and prints out each number.
-let number = 1;
+let number9 = 1;
 do {
-	console.log(number);
-	number++;
-} while (number<=10);
+	console.log(number9);
+	number9++;
+} while (number9 <= 10);
 
 console.log('- - - - - - - - - - - - - - - - - ');
 // 10. Write a for...in loop that prints out the keys and values of an object.
-const person = {name: "John", surname: "Doe", age: 30, city: "New York"};
-// Array olsaydi boyle olurdu: ["John", "Doe", 30 , "New York"];
-// Array element position numbers: 0, 1, 2, 3
-// Object keys: name, surname, age, city 
-
-for (const key in person){
-	console.log(`${key}: ${person[key]}`);
-}
+let person = {
+    name: "John",
+    surname: "Doe"
+    age: 30,
+    city: "New York"
+  };
+  
+  for (let key in person) {
+    console.log(`Key: ${key}, Value: ${person[key]}`);
+  }
+  
 console.log('- - - - - - - - - - - - - - - - - ');
+
 // 11. Write a for...in loop that sums the values of an object.
-const obj = {a: 10, b: 20, c: 30, d: 23};
-let sum = 0;
-for(const key in obj) {
-	sum+= obj[key];
-}
-console.log(sum);
+
+const myObject = {
+    value1: 10,
+    value2: 20,
+    value3: 30
+  };
+  
+  let sum = 0;
+  
+  for (let key in myObject) {
+    sum += myObject[key];
+  }
+  
+  console.log("Sum:", sum);
+  
 console.log('- - - - - - - - - - - - - - - - - ');
+
 // 12. Write a loop that finds the average of an array of numbers.
-const numbers12 = [10, 20, 30, 40, 50];
+
+const numbers = [10, 20, 30, 40, 50];
+let sumArray = 0;
+
+for (let i = 0; i < numbers.length; i++) {
+  sum += numbers[i];
+}
+
+const average = sum / numbers.length;
+console.log("Average:", averageArray);
+
 
 console.log('- - - - - - - - - - - - - - - - - ');
 // 13. Write a loop that prints out the sum of all even numbers between 1 and 50.
 
+let sum = 0;
+
+for (let i = 2; i <= 50; i += 2) {
+  sum += i;
+}
+
+console.log("Sum of even numbers:", sum);
 
 
 console.log('- - - - - - - - - - - - - - - - - ');
 // 14. Write a loop that finds the second largest number in an array of numbers.
 const numbers14 = [10, 5, 20, 8, 15];
+let largest = -Infinity;
```

Comment:
The solution for this exercise does not seem to work.

Try to start like this:
```
    let firstLargest = Math.max(numbers[0], numbers[1]);
    let secondLargest = Math.min(numbers[0], numbers[1]);
```
You'll get two first numbers of the array an assign them two the variables depending on which one of them is bigger.
Then start a loop and compare the rest of the array with the values that you assigned already. You'll need to start from the 3rd number (index 2), not from 0.